helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/Jackson2Utils.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2021 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+package com.linkedin.avroutil1.compatibility;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+
+public class Jackson2Utils {
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+  private static final JsonFactory JSON_FACTORY = new JsonFactory();
+
+  private Jackson2Utils() {
+    // Util class; should not be instantiated.
+  }
+
+  public static JsonNode toJsonNode(String str, boolean strict) {
+    if (str == null) {
+      return null;
+    }
+    try {
+      JsonParser parser = JSON_FACTORY.createParser(str);
+      JsonNode node = OBJECT_MAPPER.readTree(parser);
+      if (node == null) {
+        // Happens when str.isEmpty().
+        throw new NullPointerException("value cannot be an empty string");
+      }
+      // Calling parser.nextToken() can raise an exception itself, if there are garbage chars at the end.
+      // E.g., when str == "1,2", the ',' is the garbage char (raises JsonParseException).
+      if (parser.nextToken() != null) {
+        // Or, if that succeeds and returns a valid token, the extra chars (while not garbage) are still unwanted.
+        // E.g., when str == "1 2", the '2' is the extra token.
+        throw new IllegalStateException("value cannot have extra trailing chars; must be a single literal");
```

Comment:
same comment as aboce - can get the location() of the current token and/or the "junk" substring

Reply:
Done.